### ISSUE TITLE
🐞 Bug Report: Image Carousel not loading product gallery images in 3.9.0 (ED-9009)

### ISSUE THREAD

**Issue by webspacez** [elementor/elementor#20678]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When updating to Elementor PRO 3.9.0 elementor does lot load the product image gallery

### Steps to reproduce

Add Image Carousel to a product page
Dynamicly link product gallery. 
Then the images dont show or are not loaded. 

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt


== Server Environment ==
	Operating System: Linux
	Software: Apache/2
	MySQL version: MariaDB Server v10.3.37
	PHP Version: 7.4.33
	PHP Memory Limit: 128M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: https://dev.webspacez.com/lavayette
	Home URL: https://dev.webspacez.com/lavayette
	WP Multisite: No
	Max Upload Size: 64 MB
	Memory limit: 256M
	Max Memory limit: 512M
	Permalink Structure: /%category%/
	Language: nl
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: WebSpaceZ
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.6.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: nl_NL
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields
		Version: 6.0.5
		Author: WP Engine

	Classic Editor
		Version: 1.6.2
		Author: WordPress Contributors

	Duplicate Page
		Version: 4.4.9
		Author: mndpsingh287

	Ele Custom Skin
		Version: 3.1.7
		Author: Dudaster.com

	Ele Custom Skin Pro
		Version: 3.2.4
		Author: Dudaster.com

	Elementor
		Version: 3.9.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.8.1
		Author: Elementor.com

	Mollie Payments for WooCommerce
		Version: 7.3.4
		Author: Mollie

	Slide Anything - Responsive Content / HTML Slider and Carousel
		Version: 2.4.7
		Author: 

	Stamped.io Reviews &amp; UGC for WooCommerce
		Version: 2.3.2
		Author: Stamped.io

	UpdraftPlus - Backup/Restore
		Version: 1.22.23
		Author: UpdraftPlus.Com, DavidAnderson

	Variation Swatches for WooCommerce
		Version: 2.0.14
		Author: Emran Ahmed

	WooCommerce
		Version: 7.1.0
		Author: Automattic

	Yoast SEO
		Version: 19.11
		Author: Team Yoast


== Must-Use Plugins ==
	Elementor Safe Mode
		Version: 1.0.0
		Author: Elementor.com


== Gebruik van elementen ==
	
	archive : 1
		archive-posts : 1
		container : 2
	container : 12
		container : 32
		heading : 7
		image : 8
		posts : 1
		text-editor : 1
		theme-post-content : 1
	footer : 1
		container : 15
		heading : 4
		icon-list : 6
		image : 2
		social-icons : 1
		text-editor : 2
	header : 2
		container : 10
		nav-menu : 2
		theme-site-logo : 3
		woocommerce-menu-cart : 2
	loop : 3
		container : 4
		heading : 2
		image : 3
		text-editor : 3
		theme-post-title : 1
		wc-add-to-cart : 1
	page : 88
		button : 3
		container : 33
		heading : 16
		image : 18
		spacer : 3
	popup : 1
		container : 3
		image : 1
		text-editor : 1
	product : 10
		container : 471
		divider : 1
		heading : 83
		icon-list : 2
		image : 55
		image-carousel : 9
		menu-anchor : 3
		posts : 8
		text-editor : 102
		wc-add-to-cart : 9
		woocommerce-product-content : 1
		woocommerce-product-price : 8
		woocommerce-product-short-description : 5
		woocommerce-product-title : 9
	section : 1
		container : 1
		image : 3
	widget : 0
		heading : 3
	wp-page : 22
		button : 5
		container : 67
		form : 2
		google_maps : 1
		heading : 14
		icon : 3
		icon-box : 3
		image : 6
		posts : 5
		spacer : 1
		testimonial-carousel : 2
		text-editor : 26
		wc-elements : 1
		woocommerce-cart : 1
		woocommerce-checkout-page : 1
		woocommerce-my-account : 2
	single-page : 3
		container : 16
		divider : 2
		form : 2
		heading : 2
		image : 4
		posts : 1
		shortcode : 1
		spacer : 1
		text-editor : 2
		theme-post-content : 2


== Elementor experimenten ==
	Geoptimaliseerde DOM-uitvoer: Inactief
	Verbeterd laden van assets: Inactief
	Verbeterd laden van CSS: Inactief
	Pictogrammen inline lettertype: Actief
	Toegang verbeteringen: Inactief
	Extra aangepaste onderbrekingspunten: Inactief
	Import export template kit: Standaard actief
	Verberg verouderde WordPress widgets voor zoekresultaten: Actief
	admin_menu_rearrangement: Standaard inactief
	Flexbox container: Actief
	Default to New Theme Builder: Standaard actief
	Hello Theme header & footer: Inactief
	Landingspagina&#039;s: Standaard actief
	Kleur sampler: Inactief
	Favoriete widgets: Inactief
	Beheerder bovenste balk: Standaard actief
	Lazy-load achtergrondafbeeldingen: Standaard inactief
	Opslaan als standaard: Standaard actief
	Page Transitions: Inactief
	Notes: Inactief
	Loop: Standaard inactief
	Form Submissions: Standaard actief
	Scroll Snap: Inactief


== Log ==
	
PHP: showing 7 of 7PHP: 2022-11-11 11:43:42 [error X 2][../wp-content/plugins/elementor/core/common/modules/ajax/module.php::265] Allowed memory size of 268435456 bytes exhausted (tried to allocate 25614560 bytes) [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-11-17 11:41:28 [notice X 19][../wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-11-28 23:27:20 [notice X 6][../wp-content/plugins/elementor/core/files/css/base.php::744] Undefined index: background_overlay_color [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-11-29 00:44:15 [notice X 6][../wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-image.php::40] Undefined offset: 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-12-05 14:24:53 [notice X 1][../wp-content/plugins/elementor/includes/managers/image.php::80] Undefined offset: 2 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-12-06 13:12:10 [notice X 2][../wp-content/plugins/elementor/includes/managers/image.php::108] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-12-08 00:53:35 [error X 1][../wp-content/plugins/elementor-pro/modules/theme-builder/module.php::34] Uncaught Error: Class &#039;ElementorPro\Core\Utils&#039; not found in ../wp-content/plugins/elementor-pro/modules/theme-builder/module.php:34
Stack trace:
#0 ../wp-content/plugins/elementor-pro/modules/theme-builder/conditions/archive.php(39): ElementorPro\Modules\ThemeBuilder\Module::get_public_post_types()
#1 ../wp-content/plugins/elementor-pro/modules/theme-builder/conditions/condition-base.php(66): ElementorPro\Modules\ThemeBuilder\Conditions\Archive-&gt;register_sub_conditions()
#2 ../wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(182): ElementorPro\Modules\ThemeBuilder\Conditions\Condition_Base-&gt;__construct(Array)
#3 ../wp-content/plugins/elementor-pro/modules/theme-builder/cla [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 19 of 19JS: 2022-11-11 17:06:36 [error X 33][../wp-includes/js/jquery/jquery.min.js?ver=3.6.1:2:51824] Cannot read properties of undefined (reading \&#039;replace\&#039;) 
JS: 2022-11-12 20:51:22 [error X 9][../wp-content/plugins/elementor/assets/js/new-template.min.js?ver=3.8.0:2:2932] elementor_new_template_form_controls is not defined 
JS: 2022-11-12 21:27:25 [error X 3][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.8.0:3:883015] Cannot read properties of undefined (reading \&#039;route\&#039;) 
JS: 2022-11-12 22:39:12 [error X 5][../wp-content/plugins/elementor-pro/assets/js/editor.min.js?ver=3.8.1:3:99294] Cannot convert undefined or null to object 
JS: 2022-11-16 11:39:12 [error X 2][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.8.0:3:288289] Cannot read properties of undefined (reading \&#039;id\&#039;) 
JS: 2022-11-17 14:56:14 [error X 2][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view295\&quot;) has already been destroyed and cannot be used. 
JS: 2022-11-20 20:22:45 [error X 38][../wp-content/plugins/ultimate-elementor/assets/min-js/uael-nav-menu.min.js?ver=1.36.12:1:15477] Cannot read properties of undefined (reading \&#039;left\&#039;) 
JS: 2022-11-20 20:36:29 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.8.1:3:769459] Cannot read properties of undefined (reading \&#039;localized\&#039;) 
JS: 2022-11-28 10:23:51 [error X 1][../wp-content/plugins/elementor/assets/js/frontend.min.js?ver=3.8.1:2:30926] elementor.settings.page is undefined 
JS: 2022-11-28 10:47:36 [error X 1][../wp-includes/js/jquery/jquery.min.js?ver=3.6.1:2:51816] a is undefined 
JS: 2022-11-29 18:48:20 [error X 7][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view292\&quot;) has already been destroyed and cannot be used. 
JS: 2022-11-29 20:13:40 [error X 7][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view303\&quot;) has already been destroyed and cannot be used. 
JS: 2022-11-29 20:19:33 [error X 1][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view196\&quot;) has already been destroyed and cannot be used. 
JS: 2022-11-29 21:31:15 [error X 1][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view1815\&quot;) has already been destroyed and cannot be used. 
JS: 2022-11-29 21:31:59 [error X 1][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view5930\&quot;) has already been destroyed and cannot be used. 
JS: 2022-11-29 22:55:13 [error X 1][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view302\&quot;) has already been destroyed and cannot be used. 
JS: 2022-11-30 11:34:06 [error X 1][../wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \&quot;view235\&quot;) has already been destroyed and cannot be used. 
JS: 2022-12-05 14:50:30 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.8.1:3:649119] this.model.isValidChild is not a function 
JS: 2022-12-05 15:25:39 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.8.1:3:572731] Cannot read properties of null (reading \&#039;id\&#039;) 

Log: showing 16 of 162022-11-17 22:15:13 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.0',
  'to' => '3.8.1',
)]
2022-11-17 22:15:14 [info] elementor::elementor_updater Started 
2022-11-17 22:15:14 [info] Elementor/Upgrades - _on_each_version Start  
2022-11-17 22:15:15 [info] Elementor/Upgrades - _on_each_version Finished 
2022-11-17 22:15:15 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.0',
  'to' => '3.8.1',
)]
2022-11-23 01:14:00 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.8.1',
  'to' => '3.8.2',
)]
2022-11-23 01:14:01 [info] elementor-pro::elementor_pro_updater Started 
2022-11-23 01:14:01 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-11-23 01:14:01 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-11-23 01:14:01 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.8.1',
  'to' => '3.8.2',
)]
2022-11-23 01:14:03 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.8.1',
  'to' => '3.8.2',
)]
2022-12-07 11:42:53 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.1',
  'to' => '3.9.0',
)]
2022-12-07 11:42:53 [info] elementor::elementor_updater Started 
2022-12-07 11:42:53 [info] Elementor/Upgrades - _on_each_version Start  
2022-12-07 11:42:55 [info] Elementor/Upgrades - _on_each_version Finished 
2022-12-07 11:42:55 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.1',
  'to' => '3.9.0',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibiliteit niet gespecificeerd

== Elementor Pro - Compatibility Tag ==

````

</details>

**Comment by davidefiorio** [comment]
Also I have this problem! 
Rollback to V 3.8.1 solves it.

**Comment by nicholaszein** [comment]
Hello @webspacez and @davidefiorio! 

Thank you for your feedback! 🙏 

We are aware of this issue and a fix is on the way! 💪

Thank you for taking the time to report to us!

Best regards

**Comment by nicholaszein** [comment]
Hey @webspacez!

It seems I forgot to close this issue as solved a long time ago. My apologies. 🙏 

It was [solved with the release of Elementor Pro v3.9.1](https://elementor.com/pro/changelog/#:~:text=Fix%3A%20WooCommerce%20Product%20Gallery%20dynamic%20tag%20is%20not%20working%20in%20Basic%20Gallery%2C%20Gallery%20and%20Image%20Carousel%20widgets%20(%2320678)).

I am closing this issue now!

Thank you for reporting it!
